pkgs/flox-activation-scripts/default.nix

```diff
@@ -1,84 +1,106 @@
 {
   bash,
   coreutils,
+  daemonize,
+  fd,
   findutils,
+  flox-activations,
   getopt,
   gnused,
-  util-linux,
-  ld-floxlib,
-  runCommand,
-  shellcheck,
-  stdenv,
-  process-compose,
-  jq,
   iconv,
+  jq,
+  ld-floxlib,
   nawk,
-  fd,
-  flox-activations,
+  process-compose,
+  runCommandNoCC,
+  shellcheck,
   shfmt,
-  daemonize,
+  stdenv,
+  substituteAllFiles,
+  util-linux,
 }:
 let
-  ld-floxlib_so = if stdenv.isLinux then "${ld-floxlib}/lib/ld-floxlib.so" else "__LINUX_ONLY__";
-  ldconfig = if stdenv.isLinux then "${iconv}/bin/ldconfig" else "__LINUX_ONLY__";
-  # Some versions of Nix don't support `.` in name
+  activation-scripts = substituteAllFiles {
+    src = ../../assets/activation-scripts;
+    files = [ "." ]; # Perform recursive substitution on all files.
+    # Substitute all of the following variables.
+    inherit
+      bash
+      coreutils
+      daemonize
+      fd
+      findutils
+      getopt
+      gnused
+      jq
+      nawk
+      ;
+    # Note that substitution doesn't work with variables containing "-"
+    # so we need to create and use alternative names.
+    process_compose = process-compose;
+    flox_activations = flox-activations;
+    util_linux = util-linux;
+    # Make clear when packages are not available on Darwin.
+    ld_floxlib = if stdenv.isLinux then ld-floxlib else "__LINUX_ONLY__";
+    iconv = if stdenv.isLinux then iconv else "__LINUX_ONLY__";
+  };
```

Comment:
personally, i'd define the bin paths to imports here rather than on the bash side, so that the substituted side does not need to know which packages these tools come from.


e.g.

here:

```nix
fd = "${fd}/bin/fd"
```

and at the usage side:

```
_fd=@fd@
```

Reply:
I was trying to remain as consistent as possible with existing usage patterns to introduce the minimum change, and up until this point there were about the same number of references being passed as paths as packages, but crucially in the case of `coreutils` that was _only_ being passed as a package, and then consumed to produce a variety of paths in multiple scripts. This was very relevant because it meant that modifying a script to introduce a new coreutils program only meant modifying that single script, rather than having to modify that _and_ the `default.nix` file as well. That, and the simplicity/elegance of being able to use `inherit` from within `substituteAllFiles` made it seem that standardizing on the passing of packages was the most straightforward way to maintain this over time.

Of course I acknowledge it would work the other way as well, but would suggest for the sake of progress we revisit that in a later PR.